File: BKLastDance/src/main/java/com/thesis/projectmanagement/service/ProjectService.java
package com.thesis.projectmanagement.service;

import com.thesis.projectmanagement.dto.project.ProjectRequest;
import com.thesis.projectmanagement.dto.project.ProjectResponse;

import java.util.List;

public interface ProjectService {
    List<ProjectResponse> getAllProjects();
    ProjectResponse getProjectById(Long id);
    ProjectResponse createProject(ProjectRequest request);
    ProjectResponse updateProject(Long id, ProjectRequest request);
    void deleteProject(Long id);
}

File: BKLastDanceFE/src/services/projectService.ts

import { api } from "./api";

export interface Project {
  id: string;
  name: string;
  description?: string;
  createdAt: Date;
  updatedAt: Date;
}

export const projectService = {
  getAllProjects: async (): Promise<Project[]> => {
    const response = await api.get("/api/projects");
    return response.data;
  },
  
  getProjectById: async (id: string): Promise<Project> => {
    const response = await api.get(`/api/projects/${id}`);
    return response.data;
  },
  
  createProject: async (project: Omit<Project, "id">): Promise<Project> => {
    const response = await api.post("/api/projects", project);
    return response.data;
  },
  
  updateProject: async (id: string, project: Partial<Project>): Promise<Project> => {
    const response = await api.put(`/api/projects/${id}`, project);
    return response.data;
  },
  
  deleteProject: async (id: string): Promise<void> => {
    await api.delete(`/api/projects/${id}`);
  }
};
